pytransform3d/test/test_transform_manager.py

```diff
@@ -333,3 +338,129 @@ def test_remove_and_add_connection():
+
+
+def test_internals():
+    rng = np.random.default_rng(0)
+    A2B = random_transform(rng)
+
+    rng = np.random.default_rng(42)
+    A2C = random_transform(rng)
+
+    tm = TemporalTransformManager()
+
+    tm.add_transform("A", "B", StaticTransform(A2B))
+    tm.add_transform("A", "C", StaticTransform(A2C))
+
+    tm.remove_transform("A", "C")
+    assert ("A", "C") not in tm.transforms
+
+
+def create_sinusoidal_movement(
+    duration_sec, sample_period, x_velocity, y_start_offset, start_time
+):
+    """Create a planar (z=0) sinusoidal movement around x-axis."""
+    time_arr = np.arange(0, duration_sec, sample_period) + start_time
+    N = len(time_arr)
+    x_arr = np.linspace(0, x_velocity * duration_sec, N)
+
+    spatial_freq = 1 / 5  # 1 sinus per 5m
+    omega = 2 * np.pi * spatial_freq
+    y_arr = np.sin(omega * x_arr)
+    y_arr += y_start_offset
+
+    dydx_arr = omega * np.cos(omega * x_arr)
+    yaw_arr = np.arctan2(dydx_arr, np.ones_like(dydx_arr))
+
+    pq_arr = list()
+    for i in range(N):
+        R = active_matrix_from_extrinsic_euler_zyx([yaw_arr[i], 0, 0])
+        T = transform_from(R, [x_arr[i], y_arr[i], 0])
+        pq = pq_from_transform(T)
+        pq_arr.append(pq)
+
+    return time_arr, np.array(pq_arr)
+
+
+def test_numpy_timeseries_transform():
+    # create entities A and B together with their transformations from world
+    duration = 10.0  # [s]
+    sample_period = 0.5  # [s]
+    velocity_x = 1  # [m/s]
+    time_A, pq_arr_A = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=0.0, start_time=0.1
+    )
+    transform_WA = NumpyTimeseriesTransform(time_A, pq_arr_A)
+
+    time_B, pq_arr_B = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=2.0,
+        start_time=0.35
+    )
+    transform_WB = NumpyTimeseriesTransform(time_B, pq_arr_B)
+
+    tm = TemporalTransformManager()
+
+    tm.add_transform("A", "W", transform_WA)
+    tm.add_transform("B", "W", transform_WB)
+
+    query_time = time_A[0]  # start time
+    A2W_at_start = transform_from_pq(pq_arr_A[0, :])
+    A2W_at_start_2 = tm.get_transform_at_time("A", "W", query_time)
+    assert_array_almost_equal(A2W_at_start, A2W_at_start_2, decimal=2)
+
+    query_time = 4.9  # [s]
+    A2B_at_query_time = tm.get_transform_at_time("A", "B", query_time)
+
+    origin_of_A_pos = vector_to_point([0, 0, 0])
+    origin_of_A_in_B_pos = transform(A2B_at_query_time, origin_of_A_pos)
+    origin_of_A_in_B_xyz = origin_of_A_in_B_pos[:-1]
+
+    origin_of_A_in_B_x, origin_of_A_in_B_y = \
+        origin_of_A_in_B_xyz[0], origin_of_A_in_B_xyz[1]
+
+    assert origin_of_A_in_B_x == pytest.approx(-1.11, abs=1e-2)
+    assert origin_of_A_in_B_y == pytest.approx(-1.28, abs=1e-2)
+
+
+def test_numpy_timeseries_transform_wrong_input_shapes():
+    n_steps = 10
+    with pytest.raises(
+            ValueError, match="Number of timesteps does not equal to number "
```

Comment:
I would just match for a substring, so the essence of the error is tested, such as "number of timesteps" so we can still correct spelling or verbosity of the message... But this is probably subjective